spring-boot-tools/spring-boot-loader-tools/src/main/resources/org/springframework/boot/loader/tools/launch.script

```diff
@@ -116,9 +116,17 @@ PID_FOLDER="$PID_FOLDER/${identity}"
 pid_file="$PID_FOLDER/{{pidFilename:${identity}.pid}}"
 log_file="$LOG_FOLDER/$LOG_FILENAME"
 
-# Determine the user to run as if we are root
-# shellcheck disable=SC2012
-[[ $(id -u) == "0" ]] && run_user=$(ls -ld "$jarfile" | awk '{print $3}')
+# The application will use the user specified via
+if [[ -z "{{runUser}}" ]]; then
+    # Determine the user to run as if we are root
+    # shellcheck disable=SC2012
+    [[ $(id -u) == "0" ]] && run_user=$(ls -ld "$jarfile" | awk '{print $3}')
+else
+    run_user="{{runUser}}"
+fi
+
+# Issue an warning if the application will run as root
+[[ $(id -u) == "0" ]] && { echoRed "Application is running as root (UID 0)."; }
```

Comment:
I think this should be `echoYellow` rather than `echoRed`. The latter is only used in the script when there's an error and the script will exit with a non-zero exit code.

Reply:
And maybe also re-phrase to something like
`"[WARN] Application is running as root (UID 0). This is considered insecure."`
to emphasize that it is a warning and not just a yellow colored information :)